src/installer/corehost/common/pal.unix.cpp

```diff
@@ -331,6 +332,55 @@ bool pal::get_temp_directory(pal::string_t& tmp_dir)
     return false;
 }
 
+bool pal::get_default_bundle_extraction_base_dir(pal::string_t& extraction_dir)
+{
+    if (!get_temp_directory(extraction_dir))
+    {
+        return false;
+    }
+
+    append_path(&extraction_dir, _X(".net"));
+    pal::string_t dotnetdir(extraction_dir);
+
+    // getuid() is the real user ID, and the call has no defined errors.
+    struct passwd* passwd = getpwuid(getuid());
```

Comment:
Is there a change `getpwuid()` can be called by another thread?

Also, nit:
```suggestion
    const passwd* passwd = getpwuid(getuid());
```

(No need to say `struct` in C++, it's always implied in type declarations.)

Reply:
> Is there a change `getpwuid()` can be called by another thread?

`getpwuid()` call here is within the `get_default_bundle_extraction_base_dir()` specifically run only at apphost start. So, there's no other thread running. 

I thought of using `getpwuid_r(...)` but realized we'll have bigger problems if something else starts running before the contents are extracted.